Sample.Calculator.Simulator/SimulatorAutomation.cs

```diff
@@ -37,7 +40,7 @@ public override bool ShouldProcess()
     public override void Init(FileStatePoco poco)
     {
         base.Init(poco);
-        if (poco is CalculatorFileStatePoco mPoco)
+        if (poco is CalculatorFileStatePoco mPoco && mPoco.ModelType != null)
         {
             ModelType = mPoco.ModelType;
         }
```

Comment:
a few linces over, if you mark `public string? ModelType { get; set; }` with `?` then no need for other changes` (`ModelType = string.Empty;`  and `mPoco.ModelType != null`)

Reply:
it's an example so better to keep it simple